chef/cookbooks/crowbar-hacks/recipes/default.rb

```diff
@@ -76,26 +76,35 @@
   end
 end
 
-if node["uefi"]
-  node["uefi"]["boot"]["order"].each do |order|
-    entry = node["uefi"]["entries"][order]
-    next if entry[:active]
+ruby_block "uefi_boot_order_config" do
+  block do
+    if node["uefi"] && File.exist?("/sys/firmware/efi")
+      node["uefi"]["boot"]["order"].each do |order|
+        entry = node["uefi"]["entries"][order]
+        next if entry[:active]
 
-    Chef::Log.info("Activating UEFI boot entry #{sprintf("%x", order)}: #{entry["title"]}")
-    ::Kernel.system("efibootmgr -a -b #{sprintf('%x', order)}")
-  end
+        Chef::Log.info("Activating UEFI boot entry "\
+                       "#{sprintf("%x", order)}: #{entry["title"]}")
+        ::Kernel.system("efibootmgr --active --bootnum #{format("%x", order)}")
+      end
 
-  neworder = node["uefi"]["boot"]["order"].partition do |order|
-    node["uefi"]["entries"][order]["device"] =~ /[\/)]MAC\(/i rescue false
-  end.flatten
+      neworder = node["uefi"]["boot"]["order"].partition do |order|
```

Comment:
for the record: hound will complain (or I expect that), but in this case are spaces